### PR TITLE
Add rotate keys tx validation

### PR DESCRIPTION
**Pull request by matteojug**
## Description

Closes: https://github.com/stacks-network/sbtc/issues/663

## Changes

Add validation for rotate keys contract call.

## Testing Information

Added tests for the different error cases.

## Checklist:

- [ ] I have performed a self-review of my code
- [ ] My changes generate no new warnings
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: signer/src/storage/postgres.rs [hunk 1]
```diff
@@ -1442,6 +1443,60 @@ impl super::DbRead for PgStore {
         .map_err(Error::SqlxQuery)
     }
 
+    async fn key_rotation_exists(
+        &self,
+        chain_tip: &model::BitcoinBlockHash,
+        signer_set: &BTreeSet<PublicKey>,
+        aggregate_key: &PublicKey,
+        signatures_required: u16,
+    ) -> Result<Option<model::RotateKeysTransaction>, Error> {
```
Comment (djordon) [inline signer/src/storage/postgres.rs:1452]: nit: this should probably return a `bool`. Or maybe we should rename the function.

**Review by djordon — COMMENTED** [review]
So far so good, I need to go through the integration tests.

**Review by djordon — APPROVED** [review]
Looks good ✅!

File: signer/src/stacks/contracts.rs [hunk 2]
```diff
@@ -1107,23 +1108,128 @@ impl AsContractCall for RotateKeysV1 {
...
+        if db
+            .key_rotation_exists(
+                &req_ctx.chain_tip.block_hash,
+                &self.new_keys,
+                &self.aggregate_key,
+                self.signatures_required,
+            )
+            .await?
+            .is_some()
+        {
```
[the remark this answers is not in the document]
Comment (matteojug) [reply signer/src/stacks/contracts.rs]: Done in [3a726de](https://github.com/stacks-network/sbtc/pull/815/commits/3a726dea891cbc88b321658ef5c500cb93a6f36e)

File: signer/src/storage/postgres.rs (hunk 1 above)
Comment (matteojug) [reply signer/src/storage/postgres.rs]: Done in [3a726de](https://github.com/stacks-network/sbtc/pull/815/commits/3a726dea891cbc88b321658ef5c500cb93a6f36e)

File: signer/tests/integration/rotate_keys.rs [hunk 3]
```diff
@@ -0,0 +1,517 @@
...
+            .expect("failed to get stacks chain tip")
+            .expect("no stacks chain tip");
+
+        let mut tx = Vec::new();
+        let bitcoin_tx: BitcoinTx = fake::Faker.fake_with_rng(rng);
+        bitcoin_tx.consensus_encode(&mut tx).unwrap();
+
+        let raw_tx = Transaction {
+            txid: bitcoin_tx.compute_txid().to_byte_array(),
+            tx,
```
[the remark this answers is not in the document]
Comment (matteojug) [reply signer/tests/integration/rotate_keys.rs]: Done in [86397b9](https://github.com/stacks-network/sbtc/pull/815/commits/86397b9c8a5ca8e4237a7754fe3bee1684c925ab)

**Pull request merged**